### PR TITLE
Knob widget color ranges

### PR DESCRIPTION
**Pull request by isigmund**
Enable functionality to define color ranges for the knob widget based on the value

see also issue #52 

Signed-off-by: Ingo Sigmund <Ingo.sigmund@web.de>

### CODE REVIEW

File: web/app/widgets/knob/knob.settings.tpl.html [hunk 1]
```diff
@@ -338,6 +338,72 @@ <h4 class="col-lg-12 col-md-12">Tron skin settings</h4>
                     </div>
                 </uib-tab>
 
+                <uib-tab heading="Ranges">
+                    <br />
+                    <div class="col-lg-12 col-md-12">
+                        <div class="form-group" ng-class="{error: _form.rangesEnabled.$error && _form.submitted}">
+                            <!-- <label class="control-label col-lg-3 col-md-3">Ranges</label>  -->
```
Comment (kubawolanin) [inline web/app/widgets/knob/knob.settings.tpl.html]: Please remove commented out code

File: web/app/widgets/knob/knob.widget.js [hunk 2]
```diff
@@ -112,10 +112,38 @@
...
             }
         };
 
+
+        // if ranges are enabled update knob setings
+        if ( vm.widget.rangesEnabled) {
+          vm.widget.ranges.forEach(function(rangeItem) {
+              if (rangeItem.min != 0 || rangeItem.max != 0) {
+                  console.log("Matching:" + vm.widget.rangesTextColorMatching);
+                  if (vm.widget.rangesTextColorMatching == true) {
```
Comment (kubawolanin) [inline web/app/widgets/knob/knob.widget.js]: The only difference between these conditions is `textColor` value. 
You could simplify the statement like this:
```js
var textColor = vm.widget.textColor;

if (vm.widget.rangesTextColorMatching === true) {
    textColor = rangeItem.barColor;
}

vm.knob.options.ranges.push({
   min: rangeItem.min,
   max: rangeItem.max,
   barColor: rangeItem.barColor,
   textColor: textColor
});

```

File: web/app/widgets/knob/knob.widget.js [hunk 3]
```diff
@@ -112,10 +112,38 @@
...
+                ranges: []
             }
         };
 
+
+        // if ranges are enabled update knob setings
+        if ( vm.widget.rangesEnabled) {
+          vm.widget.ranges.forEach(function(rangeItem) {
+              if (rangeItem.min != 0 || rangeItem.max != 0) {
+                  console.log("Matching:" + vm.widget.rangesTextColorMatching);
```
Comment (kubawolanin) [inline web/app/widgets/knob/knob.widget.js]: Do we really need that `console.log` here?

Comment (isigmund) [inline web/app/widgets/knob/knob.widget.js]: no of course not - I will remove that.

File: web/app/widgets/knob/knob.widget.js (hunk 2 above)
Comment (isigmund) [inline web/app/widgets/knob/knob.widget.js]: you are right - will simplify this

**Comment by isigmund** [comment]
did the changes you requested - ready to be merged

**Comment by kubawolanin** [comment]
LGTM 👍

**Comment by ghys** [comment]
If Kuba says so. :wink:
I managed to test it and it works well!

Thanks again for your work!

**Comment by ghys** [comment]
@kaikreuzer: just thought about this, but could you advise on how the forked depencendies used solely for openHAB projects are to be managed?
In this case the knob uses a forked version of the (quite dormant) ng-knob project under my account (https://github.com/ghys/ng-knob), which I don't mind maintaining there, but wouldn't make sense to move it under the openhab organization? I don't see any forks of other projects there but I figured it couldn't hurt to ask.

**Pull request merged**